aeneas/src/mach/GlobalRegAlloc.v3

```diff
@@ -0,0 +1,603 @@
+// Copyright 2023 Virgil Authors. All rights reserved.
+// See LICENSE for details of Apache 2.0 license.
+
+def verbose = false;
+def colorVerbose = false;
+// Implement global register allocation with graph coloring. Build an interference
+// graph from liveness analysis.
+class GlobalRegAlloc(regSet: MachRegSet, codegen: SsaMachGen) {
+	def regState = Array<int>.new(regSet.physRegs);         // indices into {allocated}
```

Comment:
The `regState` data structure seems very similar to that used in `LocalRegAlloc`. We can maybe factor it out in a separate PR.

Reply:
Sure. After merging #143, I will modify `GlobalRegAlloc`.